pkg/web/src/app/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx

```diff
@@ -301,44 +331,112 @@ export const AddEditProviderModal: React.FunctionComponent<IAddEditProviderModal
                 {fields?.password ? (
                   <ValidatedPasswordInput
+                    inputProps={{ onFocus: scrollVerifyButtonIntoView }}
                     field={fields.password}
                     label={getLabelName('pwd', brandPrefix(providerType))}
                     isRequired
                     fieldId="password"
                   />
                 ) : null}
                 {fields?.fingerprint ? (
-                  <ValidatedTextInput
-                    field={fields.fingerprint}
-                    label="vCenter SHA-1 fingerprint"
-                    isRequired
-                    fieldId="fingerprint"
-                    formGroupProps={{
-                      labelIcon: (
-                        <Popover
-                          bodyContent={
-                            <div>
-                              See{' '}
-                              <a href={PRODUCT_DOCO_LINK.href} target="_blank" rel="noreferrer">
-                                {PRODUCT_DOCO_LINK.label}
-                              </a>{' '}
-                              for instructions on how to retrieve the fingerprint.
-                            </div>
-                          }
-                        >
-                          <Button
-                            variant="plain"
-                            aria-label="More info for SHA-1 fingerprint field"
-                            onClick={(e) => e.preventDefault()}
-                            aria-describedby="fingerprint"
-                            className="pf-c-form__group-label-help"
-                          >
-                            <HelpIcon noVerticalAlign />
-                          </Button>
-                        </Popover>
-                      ),
-                    }}
-                  />
+                  <>
+                    {!isCertificateQueryEnabled ? (
+                      <Button
+                        id="certificate-confirm-button"
+                        key="confirm"
+                        aria-label="Verify Certificate"
+                        variant="primary"
+                        isDisabled={!fields.hostname?.isTouched || !fields.hostname?.isValid} // TODO we should remove the isTouched case here once we resolve https://github.com/konveyor/lib-ui/issues/82
+                        onClick={() => {
+                          setCertificateQueryEnabled(true);
+                        }}
+                      >
+                        Verify certificate
+                      </Button>
+                    ) : certificateQuery.status === 'loading' ? (
+                      <div className="pf-c-empty-state__icon">
+                        <Spinner
+                          aria-labelledby="loadingPrefLabel"
+                          size="sm"
+                          className={spacing.mrSm}
+                        />
+                        Retrieving SHA-1 certificate fingerprint
```

Comment:
This could maybe also be an [inline plain](https://www.patternfly.org/v4/components/alert#inline-plain-types) static live region info alert, if you wanted to communicate to screen reader users the loading state - since this one can take a while to resolve if there are issues with the network request.